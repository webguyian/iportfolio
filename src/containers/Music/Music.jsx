import React from 'react';
import { object } from 'prop-types';

import TabNavigation from 'components/TabNavigation/TabNavigation';
import MusicLibrary from './MusicLibrary/MusicLibrary';
import MusicPlayer from './MusicPlayer/MusicPlayer';

import { useInitialRoute } from 'modules/browser/hooks';

const Music = props => {
  const { match } = props;
  const baseClass = 'music-app';
  const tabs = [
    {
      path: `${match.path}/library`,
      icon: 'music',
      label: 'Library',
      component: MusicLibrary
    },
    {
      path: `${match.path}/player`,
      icon: 'play-circle',
      label: 'Player',
      component: MusicPlayer
    }
  ];

  useInitialRoute(match, tabs[0].path);

  return (
    <div className={baseClass}>
      <TabNavigation tabs={tabs} />
    </div>
  );
};

Music.propTypes = {
  match: object.isRequired
};

export default Music;
