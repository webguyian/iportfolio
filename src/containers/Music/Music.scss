.music-app {
  @include stretch-column-nowrap;
  padding-top: $gutter;

  .ui-tab-navigation-link--active {
    > .ui-icon,
    > .ui-tab-navigation-link-label {
      color: $pink;
    }
  }
}
