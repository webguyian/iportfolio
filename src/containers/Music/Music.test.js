import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

import Music from './Music';

describe('<Music />', () => {
  const props = {
    match: {
      path: '/music'
    }
  };

  it('renders correctly', () => {
    const component = render(
      <MemoryRouter>
        <Music {...props} />
      </MemoryRouter>
    );

    expect(component.asFragment()).toMatchSnapshot();
  });
});
