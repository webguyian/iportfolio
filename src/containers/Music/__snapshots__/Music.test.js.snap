// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Music /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="music-app"
  >
    <nav
      class="ui-tab-navigation"
    >
      <ul
        class="ui-tab-navigation-list"
      >
        <li
          class="ui-tab-navigation-list-item"
        >
          <a
            class="ui-tab-navigation-link ui-link"
            href="/music/library"
          >
            <i
              class="ui-icon ui-icon-music fas fa-music fa-3x"
            />
            <span
              class="ui-text ui-text--body ui-tab-navigation-link-label"
            >
              Library
            </span>
          </a>
        </li>
        <li
          class="ui-tab-navigation-list-item"
        >
          <a
            class="ui-tab-navigation-link ui-link"
            href="/music/player"
          >
            <i
              class="ui-icon ui-icon-play-circle fas fa-play-circle fa-3x"
            />
            <span
              class="ui-text ui-text--body ui-tab-navigation-link-label"
            >
              Player
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
